spec/abstract-jobs.html

```diff
@@ -91,12 +184,13 @@ <h1> CheckForEmptyCells ( _finalizationGroup_ ) </h1>
     </emu-alg>
   </emu-clause>
 
-  <emu-clause id="sec-finalization-group-cleanup-job" aoid="FinalizationGroupCleanupJob">
-    <h1> FinalizationGroupCleanupJob ( _finalizationGroup_ [ , _callback_ ] ) </h1>
+  <emu-clause id="sec-cleanup-finalization-group" aoid="CleanupFinalizationGroup">
+    <h1> CleanupFinalizationGroup ( _finalizationGroup_ [ , _callback_ ] ) </h1>
     <p> The following steps are performed: </p>
     <emu-alg>
       1. Assert: _finalizationGroup_ has [[Cells]],
       [[CleanupCallback]], and [[IsFinalizationGroupCleanupJobActive]] internal slots.
+      1. If CheckForEmptyCells(_finalizationGroup_) is *false*, return.
```

Comment:
I just realized a side-effect of this PR. Before, calling `finalizationGroup.cleanupSome()` would always invoke callback even when no cells were empty. The iterator would simply not yield anything.
After the change, the callback is not invoked if the finalization group doesn't have any empty cell.

Just wanted to double check the new behavior was intended.

Reply:
This might make some implementations harder: now the finalization group either needs to check all cells immediately when `cleanupSome` is called instead of relying on the iterator use by the caller, or it need to internally keep track if it has an empty cell.

Reply:
Yes, this line of this PR does make that change. Should we remove it?

Reply:
I don't think this is such a big burden, since something will have to be found when looking for the .next() entry anyway

Reply:
It make some of my tests more complicated ;)\
So far I was relying on calling `cleanupSome` to get access to the iterator. Now I can't do that anymore without forcing a gc.

I just rewrote my shim to include that new behavior, and fix the scheduling of "host" jobs.\
Keeping track of empty cells allowed me to avoid iterating blindly over all cells in the cleanup job. Not sure how real engines would implement this, but in my case I wouldn't consider it a big burden.

Reply:
OK. I think we should leave the semantics as is for now, but if we hear about more problems, we can consider reverting it.